crates/local-cluster-runner/src/node/mod.rs

```diff
@@ -809,7 +809,19 @@ impl StartedNode {
             return false;
         };
 
-        response.status() == MetadataServerStatus::Member
+        if response.status() != MetadataServerStatus::Member {
+            return false;
+        }
+
+        let configuration = response.configuration.expect("must be set");
+        if configuration.members.len() != cluster_size {
```

Comment:
Nice! Is it worth it doing something similar for the number of provisioned logs and active partition processors to fully close the loop on provisioning status, too? So, fetch the logs config and partition state, and confirm we've got the capacity we expect. An assertion that number of active partition processor = configured partitions would capture both.